src/analyses/accessAnalysis.ml

```diff
@@ -43,7 +43,7 @@ struct
       + [deref=true], [reach=true] - Access [exp] by dereferencing transitively (reachable), used for deep special accesses. *)
   let access_one_top ?(force=false) ?(deref=false) ctx (kind: AccessKind.t) reach exp =
     if M.tracing then M.traceli "access" "access_one_top %a %b %a:\n" AccessKind.pretty kind reach d_exp exp;
-    if force || !collect_local || !emit_single_threaded || ThreadFlag.is_multi (Analyses.ask_of_ctx ctx) then (
+    if force || !collect_local || !emit_single_threaded || ThreadFlag.has_ever_been_multi (Analyses.ask_of_ctx ctx) then (
```

Comment:
```suggestion
    if force || !collect_local || !emit_single_threaded || ThreadFlag.is_currently_multi (Analyses.ask_of_ctx ctx) then (
```
Because accesses and races only depend on the current state. Or is there some reason for this?

Reply:
The access event needs to be emitted here, such that protection is computed in a way that coincides with the way the privatizations expect it to be computed.
This does only impact protection but race checking still works!